platform/ext/target/nordic_nrf/common/spm_hal.c

```diff
@@ -321,3 +321,68 @@ enum tfm_plat_err_t tfm_spm_hal_nvic_interrupt_enable(void)
 {
     return nvic_interrupt_enable();
 }
+
+bool tfm_spm_hal_has_access_to_region(const void *p, size_t s,
+                                              int flags)
+{
```

Comment:
Did you consider moving the implementation of this function entirely to `spu.c` (and just calling it from here)? This would reduce the code footprint and execution time by eliminating the need of several `spu_regions_*` calls.
I am wondering if it will be actually beneficial to have this whole `spu_regions_*` API. Do you foresee some other use cases for it?

Reply:
Thing is, I recall, we wanted the spu.c to be, eventually, an nRFX driver for the system protection unit exposing a relevant API, and should not contain CMSE checks that are tight to the ARM Core. That's why I favor this separation, in the first place.

If we decide to move spu. to nrfx, we cannot have functions that use the TT intrinsics, or other v8-m or CMSE functionality.

Reply:
(Also, strictly, speaking, we cannot call the spu.c API from tfm-core; there _has_ to do a platform wrapper function, at least a transparent wrapper.)

Reply:
> If we decide to move spu. to nrfx, we cannot have functions that use the TT intrinsics, or other v8-m or CMSE functionality.

I don't see why we cannot, actually. We already use some ARM specific things (like barriers) in drivers that need it.

> (Also, strictly, speaking, we cannot call the spu.c API from tfm-core; there _has_ to do a platform wrapper function, at least a transparent wrapper.)

I was thinking about a wrapper. That's why I wrote "(and just calling it from here)".

Reply:
In my opinion, the implementation should not part of the SPU driver. I don't want to have a function inside the SPU driver that assesses memory accesses, _not_ explicitly based on SPU, but based on MPU as well, and based on other TrustZone configuration, e..g the SAU ALL_NS bit flag. The function is beyond the responsibility of the SPU peripheral, and, therfore, it has to be outside the driver, in some common file, like the spm_hal.